media-video/pipewire/pipewire-0.3.51-r1.ebuild

```diff
@@ -248,12 +254,26 @@ multilib_src_install_all() {
 		dosym ../../../usr/share/alsa/alsa.conf.d/99-pipewire-default-hook.conf /etc/alsa/conf.d/99-pipewire-default-hook.conf
 	fi
 
+	# Enable required wireplumber alsa and bluez monitors
+	if use sound-server; then
+		dodir /etc/wireplumber/main.lua.d
+		echo "alsa_monitor.enabled = true" > ${D}/etc/wireplumber/main.lua.d/89-gentoo-sound-server-enable-alsa-monitor.lua
```

Comment:
Attention, main.lua.d no longer exists in the work in progress origin/moving-lua-scripts branch of WirePlumber! I can't say when that branch will get merged into the master or what shape the final implementation will take but be aware that the oncoming JSON config files feature will likely break this particular method of enabling/disabling the backends.

Reply:
No problem with that, I believe pipewire maintainers will be able to easily hit that moving target or simplify this configuration method.

Reply:
Let me repeat, the main.lua.d looks to be going away. Whatever comes after will not even be written in Lua but JSON. And as I pointed out later, the override could easily be done inside the package(s) that provide PulseAudio daemon rather than the unrelated PipeWire and WirePlumber which have nothing to do with PulseAudio daemon.

Reply:
Hehe let me repeat this too, I stated this proposed method works with exact package versions `pipewire-0.3.51-r1` and `wireplumber-0.4.10-r1`

I hope future `wireplumber` will be able to provide support for similar functionality. Main idea is that
1. you are running `wireplumber` and `pipewire-pulse` from `pipewire` start script on openrc
2. so you have to operate similar on/off switch for `pipewire-pulse` from within `pipewire` ebuild
3. and it makes sense to me to be able to operate on/off switch of audio server using `pipewire` ebuild, e.g. by augmenting installed `wireplumber` configuration to enable required parts based on `pipewire` USE `sound-server`

Reply:
My point is there is not much room for choice - one cannot freely mix PipeWire and PulseAudio daemons without sufficient understanding of how the audio stack works. And such people have the know how and ability do these modifications themselves.

In light of this, it's much simpler to just have basic logic such as `if media-sound/pulseaudio-daemon, then PipeWire audio is always disabled by the way of a drop-in file installed by media-sound/pulseaudio-daemon package`. Those who know better, know how to write a drop-in that will und othe change or they can use INSTALL_MASK or some other method to get their way.

Reply:
Again no, this is not only about making a conflict pulseaudio daemon, but about the ability to switch on/off the sound server parts of pipewire which handle alsa and bluetooth devices.